system-addon/lib/FaviconFeed.jsm

```diff
@@ -50,15 +54,20 @@ this.FaviconFeed = class FaviconFeed {
     let {data, etag, status} = await this.loadFromURL(this.prefs.get("tippyTop.service.endpoint"), headers);
     if (status === 304) {
       // Not modified, we are done. No need to dispatch actions or update cache. We'll check again in 24+ hours.
-      this.tippyTopLastUpdated = Date.now();
-      return;
-    }
-    if (status === 200 && data) {
+      this.tippyTopNextUpdate = Date.now() + TIPPYTOP_UPDATE_TIME;
+      this.numRetries = 0;
+    } else if (status === 200) {
       let sitesByDomain = this._sitesArrayToObjectByDomain(data);
       sitesByDomain._etag = etag;
-      this.tippyTopLastUpdated = sitesByDomain._timestamp = Date.now();
+      sitesByDomain._timestamp = Date.now();
       this.cache.set("sites", sitesByDomain);
       this._sitesByDomain = sitesByDomain;
+      this.tippyTopNextUpdate = sitesByDomain._timestamp + TIPPYTOP_UPDATE_TIME;
+      this.numRetries = 0;
+    } else {
+      // Something didn't go quite right. Retry with an exponential backoff.
+      let delay = Math.max(TIPPYTOP_UPDATE_TIME, TIPPYTOP_RETRY_DELAY * Math.pow(2, this.numRetries++));
```

Comment:
I'm thinking you wanted a `Math.min` to cap it at 24 hours of `delay`?

Reply:
lol. yes. #fail